witness/src/Dscp/Resource/Keys/Functions.hs

```diff
@@ -112,16 +115,27 @@ genStore comParamsM = do
 
 -- | Read store under given path.
 readStore
-    :: (MonadIO m, MonadCatch m, MonadLogging m)
+    :: (MonadIO m, MonadCatch m, MonadLogging m, MonadThrow m)
     => FilePath -> PassPhrase -> m (KeyResources n)
 readStore path pp = do
     logDebug $ "Reading key from: " +|| path ||+ ""
     content <- rewrapKeyIOErrors $ do
-        whenPosix $ ensureModeIs mode600 path
+        whenPosix $ checkFileMode mode600 path
         liftIO $ LBS.readFile path
     Versioned mid <- eitherDecode' @KeyfileContent content
         & leftToThrow (SecretParseError . toText)
     mkStore <$> fromSecretJson pp mid
+  where
+    checkFileMode
+      :: (IsPosix, MonadIO m, MonadLogging m, MonadThrow m)
+      => FileMode -> FilePath -> m ()
```

Comment:
This should go to `Dscp.System.FileMode`. Also, `MonadLogging` is an unnecessary constraint here.